src/client/Client.js

```diff
@@ -0,0 +1,103 @@
+const Q = require('q');
+const { format } = require('url');
+const request = Q.denodeify(require('request'));
+
+class MockClient {
+
+	constructor (port, serverHost) {
+		this.port = port;
+		this.serverHost = serverHost;
+	}
+
+	isListening () {
+		return this.getListener(this.port)
+			.then(response => response.statusCode === 200);
+	}
+
+	getListener () {
+		return this.req({
+			uri: '/listener/' + this.port,
+			method: 'GET'
+		});
+	}
+
+	listen () {
+		return this.req({
+			uri: '/listener/' + this.port,
+			method: 'POST'
+		});
+	}
+
+	getRequests () {
+		return this.req({
+			uri: '/requests/' + this.port,
+			method: 'GET'
+		})
+		.then(res => JSON.parse(res.body));
+	}
+
+	waitForRequest (predicate, count = 1, delay = 500) {
+		return this.getRequests()
+			.then(requests => requests.filter(predicate))
+			.then(requests => {
+				if (requests.length === count) {
+					return requests;
+				} else {
+					return Q.delay(delay).then(() => this.waitForRequest(predicate, count, delay));
+				}
+			});
+	}
+
+	removeListener () {
+		return this.req({
+			uri: '/listener/' + this.port,
+			method: 'DELETE'
+		});
+	}
+
+	clear () {
+		return this.req({
+			uri: '/clear',
+			method: 'DELETE'
+		});
+	}
+
+	allowAll () {
```

Comment:
we probably should rename this to `mockAllGetRequests` or remove this at all. It does not seem very useful